### PR TITLE
fixing hexdec on a 16 char hex number, which could actually exceed INT64

### PR DESCRIPTION
**Pull request by dragoscirjan**
On a 16 'digits' hex chunck, i.e. `bea15bfc5e308b1a`, `hexdec` method can trigger erros, because the converted number could exceed PHP integer side. 

A solution would be to create a _hexdec method using [php-bc](http://php.net/manual/en/book.bc.php) or [php-gpm](http://php.net/manual/en/book.gmp.php) to convert the chunk to a decimal number, then convert the number to a 64 bit signed INT number and parse it with `intval`.

This is my approach, it's not the best, so hoping you guys will refine it further :)

### CODE REVIEW

**Comment by dragoscirjan** [comment]
Need to fix some things before getting back with this PR.

File: src/Jaeger/SpanConverter.php [hunk 1]
```diff
@@ -125,15 +125,35 @@ private static function convertTimestamp(\DateTimeInterface $dateTime)
...
+    }
+
     /**
      * Split the provided hexId into 2 64-bit integers (16 hex chars each).
      * Returns array of 2 int values.
      */
     private static function convertTraceId($hexId)
     {
+        $method = '';
+        if (self::$preferredBigMathExt === self::BIG_MATH_BC) {
```
Comment (chingor13) [inline src/Jaeger/SpanConverter.php]: Can we turn this into an interface with 2 separate implementations and let us not have to do this check every time? We can also potentially detect the default implementation based on which extensions are loaded.

File: src/Installer.php [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace OpenCensus\Trace\Exporter;
+
+class Installer
+{
+    public static function checkPhpExtDependency()
+    {
+        if (!function_exists('bcadd') && !function_exists('gmp_add')) {
```
Comment (chingor13) [inline src/Installer.php]: Let's use `extension_loaded` here - a user *could* have `bcadd` or `gmp_add` implemented somewhere in the function lookup chain and not the rest.

**Review by chingor13 — CHANGES_REQUESTED** [review]
Thanks for submitting this!

**Comment by dragoscirjan** [comment]
Hi @chingor13, 
Please see code changes for the suggestions you have made. I think this is the best form this could be done, for now.

File: src/JaegerExporter.php [hunk 3]
```diff
@@ -76,9 +76,10 @@ public function __construct($serviceName, array $options = [])
         ];
         $this->host = $options['host'];
         $this->port = (int) $options['port'];
+        $this->spanConverter = empty($options['spanConverter']) ? new SpanConverter() : $options['spanConverter'];
```
Comment (chingor13) [inline src/JaegerExporter.php:84]: 1. Is there a need for the span converter to be injected? It's not being used for tests.

2. This needs an instance variable declaration above.

File: src/Jaeger/SpanConverterInterface.php [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+use OpenCensus\Trace\SpanData;
+
+use Jaeger\Thrift\Span;
+
+/**
+ * This class handles converting from the OpenCensus data model into its
+ * Jaeger Thrift representation.
+ */
+interface SpanConverterInterface
```
Comment (chingor13) [inline src/Jaeger/SpanConverterInterface.php]: I don't think this interface is needed. There's only one implementation and its internal (or probably should be) until the need arises for a separate implementation.

File: src/Jaeger/HexdecConverter.php [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenCensus\Trace\Exporter\Jaeger;
+
+use OpenCensus\Trace\Exporter\Jaeger\HexdecConverterInterface;
+
+class HexdecConverter implements HexdecConverterInterface
```
Comment (chingor13) [inline src/Jaeger/HexdecConverter.php:26]: Please add a class description phpdoc.

File: src/Jaeger/HexdecConverterBcMath.php [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenCensus\Trace\Exporter\Jaeger;
+
+use OpenCensus\Trace\Exporter\Jaeger\HexdecConverterInterface;
+
+class HexdecConverterBcMath implements HexdecConverterInterface
```
Comment (chingor13) [inline src/Jaeger/HexdecConverterBcMath.php:26]: Please add a class description phpdoc.

File: src/Jaeger/HexdecConverterInterface.php [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenCensus\Trace\Exporter\Jaeger;
+
+ /**
+  *
```
Comment (chingor13) [inline src/Jaeger/HexdecConverterInterface.php]: Please fill this out.

**Review by chingor13 — COMMENTED** [review]
Thank you! The refactoring looks good so far. Just added a few smallish things.

**Review by chingor13 — APPROVED** [review]
LGTM. Thanks!

**Comment by un000** [comment]
@chingor13 could you make a release for this PR?

**Pull request merged**